core/src/main/scala/org/http4s/Headers.scala

```diff
@@ -89,17 +84,31 @@ final class Headers private (headers: List[Header])
         }
 
         val h = new Headers(acc.prependToList(hs))
-        h.asInstanceOf[That]
+        h
       }
-    } else super.++(that)
+  
+  def filterNot(f: Header => Boolean): Headers =
+    Headers(headers.filterNot(f))
+
+  def filter(f: Header => Boolean): Headers =
+    Headers(headers.filter(f))
+
+  def collectFirst[B](f: PartialFunction[Header,B]): Option[B] = 
+    headers.collectFirst(f)
+
+  def foldMap[B: Monoid](f: Header => B): B =
```

Comment:
Was there any principle to which of these forwarders was included and which were not?  The choices seem reasonable, but I can't think off the top of my head what made the cut and what didn't and why.

Reply:
What was being used combined with what I consider the reasonable methods. I went for foldableLike but fixed on the base type.

Reply:
Is there a `FoldableLike` somewhere?  I've thought of the concept a few times and then remembered, "duh, Foldable is F[_]`".

Whatever, I think you're on the right track.